packages/client/src/Config.ts

```diff
@@ -26,9 +25,113 @@ export interface PrivateKeyAuthConfig {
     address?: string
 }
 
-export interface TrackerRegistryContract {
-    jsonRpcProvider?: ConnectionInfo
-    contractAddress: string
+export interface Layer0Config {
+
+    /**
+     * The list of entry point PeerDescriptors used to join the Streamr Network.
+     */
+    entryPoints?: JsonPeerDescriptor[]
+
+    /**
+     * The list of STUN and TURN servers to use in ICE protocol when
+     * forming WebRTC connections.
+    */
+    iceServers?: IceServer[]
+
+    /**
+     * When set to true private addresses will not be probed when forming
+     * WebRTC connections.
+     *
+     * Probing private addresses can trigger false-positive incidents in
+     * some port scanning detection systems employed by web hosting
+     * providers. Disallowing private addresses may prevent direct
+     * connections from being formed between nodes using IPv4 addresses
+     * on a local network.
+     *
+     * Details: https://github.com/streamr-dev/network/wiki/WebRTC-private-addresses
+    */
+    webrtcDisallowPrivateAddresses?: boolean
+
+    /**
+     * Defines WebRTC connection establishment timeout in milliseconds.
+     *
+     * When attempting to form a new connection, if not established within
+     * this timeout, the attempt is considered as failed and further
+     * waiting for it will cease.
+    */
+    newWebrtcConnectionTimeout?: number
+
+    /**
+     * Sets the low-water mark used by send buffers of WebRTC connections.
+    */
+    webrtcDatachannelBufferThresholdLow?: number
+
+    /**
+     * Sets the high-water mark used by send buffers of WebRTC connections.
+    */
+    webrtcDatachannelBufferThresholdHigh?: number
+
+    /**
+     * Contains connectivity information to the client's Network Node, used in the network layer.
+     * Can be used in cases where the client's public IP address is known before
+     * starting the network node. If not specified, the PeerDescriptor will be auto-generated.
+    */
+    peerDescriptor?: JsonPeerDescriptor
+
+    /**
+     * The port to use for the client's Network Node WebSocket server.
+     * If not specified, the server will not be started
+     */
+    webSocketPort?: number
```

Comment:
Typically we've used this form:

```suggestion
    websocketPort?: number
```

(change also `webSocketPort` and `webSocketHost` in `ConnectionManagerConfig`)

Reply:
WebSocket (web socket)

Reply:
Yes, there is no right or wrong format (e.g. `realtime`/`real_time`). But the public API of client and broker should be consistent and therefore use the `websocketPort` form. See also `JsonPeerDescriptor`.